include/seqan3/alignment/matrix/detail/alignment_score_matrix_one_column_base.hpp

```diff
@@ -0,0 +1,64 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::alignment_score_matrix_one_column_base.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <array>
+#include <utility>
+#include <vector>
+
+#include <seqan3/core/concept/core_language.hpp>
+#include <seqan3/core/simd/concept.hpp>
+#include <seqan3/range/container/aligned_allocator.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief A base class for alignment score matrices using only one column to compute the matrix.
+ * \tparam score_t The type of the scores.
+ * \ingroup alignment_matrix
+ *
+ * \details
+ *
+ * Manages the actual storage as a std::vector. How much memory is allocated is handled by the derived type.
+ * The `score_t` must either model the seqan3::Arithmetic or seqan3::detail::Simd vector concept.
+ */
+template <typename score_t>
+struct alignment_score_matrix_one_column_base
+{
+    static_assert(Arithmetic<score_t> || Simd<score_t>,
+                  "Score type must either be either an arithmetic type or a simd vector type.");
+
+    /*!\name Associated types
+     * \{
+     */
+    //!\brief The actual element type.
+    using element_type = std::tuple<score_t, score_t>;
+    //!\brief The allocator type.
+    using allocator_type = aligned_allocator<element_type, sizeof(element_type)>;
+    //!\brief The type of the underlying storage.
+    using pool_type = std::vector<element_type, allocator_type>;
+    //!\brief The size type.
+    using size_type = size_t;
+    //!\}
+
+    //!\brief The linearised memory pool storing only one column of the matrix.
+    pool_type pool{};
+    //!\brief Internal cache for the last diagonal and vertical value during the alignment computation.
```

Comment:
in the banded case I need to cache the diagonal in between. that's where I use the third parameter

Reply:
pls add that in the comment

Reply:
I did already 😉

Reply:
I don't understand. If you want to know what is the variable for you can look into the docs. If you want to touch the file and change some of it's implementation you have to look into the code anyway. So for the docs I think it is sufficient to know that it is some cached variable. To understand it's usage one needs to look into the code. And there it will be even further described.